quil-rs/src/quil.rs

```diff
@@ -0,0 +1,68 @@
+/// A trait to wrap items which represent some construct within the Quil language.
+///
+/// If you want to serialize an object to string and fail if it can't be represented as valid Quil, then use
+/// `to_quil()`. If you want to serialize an object to string infallibly, and can tolerate invalid Quil, then
+/// use `to_quil_or_debug()`.
+pub trait Quil: std::fmt::Debug {
```

Comment:
We should consider using a [sealed trait](https://rust-lang.github.io/api-guidelines/future-proofing.html) to prevent issues with library consumers implementing faulty versions of `Quil` for their custom types. The [`sealed`](https://crates.io/crates/sealed) crate seems popular, or it can be done manually

Reply:
Will consider and come back to this.